Update rosdep_repo_check targets.

nuclearsandwich opened the pull request:
* Add Debian Bullseye.
* Add Ubuntu Jammy.

nuclearsandwich committed:
Update rosdep_repo_check targets.

* Add Debian Bullseye.
* Add Ubuntu Jammy.

cottsay committed:
Merge branch 'master' into nuclearsandwich/update-repo-check-targets

cottsay commented:
The tests for this PR failed due to the issues discussed here: https://github.com/ros/rosdistro/issues/31439#issuecomment-994093767
I merged from master to work around them.

cottsay commented on test/rosdep_repo_check/config.yaml:
```diff
@@ -54,6 +54,7 @@ supported_versions:
   alpine:
   - edge
   debian:
+  - bullseye
   - buster
```
The 'latest' release of a platform should appear last. It's the one that suggestions are made for.
```suggestion
  - buster
  - bullseye
```

nuclearsandwich replied:
Thanks. I was following the rosdep requirements/convention that definitions be in lexical order but I like release order better so I'm happy to accept this change and it's good to know that the suggested platform is the last one.

nuclearsandwich committed:
Use release order rather than lexical order.

The 'latest' release of a platform should appear last. It's the one that suggestions are made for.

Co-authored-by: Scott K Logan <logans@cottsay.net>
